python/GafferUI/SpreadsheetUI/_RowsPlugValueWidget.py

```diff
@@ -276,3 +303,29 @@ def __updateVisibleSections( self ) :
 		self.__cellsTable.setVisibleSection( section )
 
 GafferUI.PlugValueWidget.registerType( Gaffer.Spreadsheet.RowsPlug, _RowsPlugValueWidget )
+
+# Signal with custom result combiner to prevent bad
+# slots blocking the execution of others.
+class _AddButtonMenuSignal( Gaffer.Signal2 ) :
+
+	def __init__( self ) :
+
+		Gaffer.Signal2.__init__( self, self.__combiner )
+
+	@staticmethod
+	def __combiner( results ) :
+
+		while True :
+			try :
+				next( results )
+			except StopIteration :
+				return
+			except Exception as e :
+				# Print message but continue to execute other slots
+				IECore.msg(
+					IECore.Msg.Level.Error,
+					"Spreadsheet Add Button menu", traceback.format_exc()
```

Comment:
Are we missing an `import traceback`?

Reply:
Doh! Sorry yes, right you are. Added [here](https://github.com/GafferHQ/gaffer/pull/4127/files#diff-0e1b620a21226cc6afd7c603dee6d8d447be54812b42404c718dbb0ad29591d2R38). Thanks John!